src/VirtualClient/VirtualClient.Actions/MLPerfTraining/MLPerfTrainingExecutor.cs

```diff
@@ -0,0 +1,517 @@
+            string checkpointPath = this.PlatformSpecifics.Combine("/mlperftraining0", $"{this.DataPath}", "mlperf-training-package", "phase1");
+
+            using (BackgroundOperations profiling = BackgroundOperations.BeginProfiling(this, cancellationToken))
+            {
+                string execCommand = $"su -c \"source {this.ConfigFile}; " + 
+                                     $"env BATCHSIZE={this.BatchSize} " + 
+                                     $"DGXNGPU={this.GPUNum} " + 
+                                     $"CUDA_VISIBLE_DEVICES=\"{this.GetGPULabels()}\" " + 
+                                     $"CONT={this.GetContainerName()} DATADIR={shardsPath} DATADIR_PHASE2={shardsPath} EVALDIR={evalPath} CHECKPOINTDIR={checkpointPath} CHECKPOINTDIR_PHASE1={checkpointPath} ./run_with_docker.sh\"";
+
+                using (IProcessProxy process = await this.ExecuteCommandAsync("sudo", execCommand, this.ExecutionDirectoryPath, telemetryContext, cancellationToken)
+                   .ConfigureAwait())
+                {
+                    if (!cancellationToken.IsCancellationRequested)
+                    {
+                        await this.LogProcessDetailsAsync(process, telemetryContext)
+                            .ConfigureAwait();
+
+                        process.ThrowIfErrored<WorkloadException>(errorReason: ErrorReason.WorkloadFailed);
+
+                        await this.CaptureMetricsAsync(process, telemetryContext, cancellationToken, MLPerfTrainingExecutor.AccuracySummary)
+                            .ConfigureAwait();
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Parse metrics and push to telemetry
+        /// </summary>
+        /// <param name="process">Execute process for StandardOutput containing the metrics</param>
+        /// <param name="telemetryContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        private async Task CaptureMetricsAsync(IProcessProxy process, EventContext telemetryContext, CancellationToken cancellationToken, string context = null)
+        {
+            // Convert StandardOutput to string
+            ConcurrentBuffer buffer = process.StandardOutput;
+
+            string logs = string.Concat(buffer.ToString(), Environment.NewLine);
+
+            await this.LogProcessDetailsAsync(process, telemetryContext, "MLPerf Training", results: logs.AsArray(), logToFile: true);
+
+            MLPerfTrainingMetricsParser parser = new MLPerfTrainingMetricsParser(logs);
+            IList<Metric> metrics = parser.Parse();
+
+            this.Logger.LogMetrics(
+                "MLPerf Training",
+                this.Scenario,
+                process.StartTime,
+                process.ExitTime,
+                metrics,
+                "Executing",
```

Comment:
Why are we keeping it as Executing, what does this signify?

Reply:
I have changed it to GPU. It is metric categorization, it is being applied on the GPU, so I am changing it to GPU